src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -882,6 +882,10 @@ public function getDefinition($id)
             throw new InvalidArgumentException(sprintf('The service definition "%s" does not exist.', $id));
         }
 
+        if ($this->definitions[$id]->isDeprecated()) {
```

Comment:
getting the definition of a deprecated service is perfectly valid and should not trigger a deprecation. The deprecation should be thrown when *instantiating* a deprecated service (in ``createService``)